### PR TITLE
LG-2669: MenuItem properly utilizes `as` prop

### PR DESCRIPTION
**Pull request by brookescarlett**
## ✍️ Proposed changes

- Previously MenuItem and SubMenu components were not properly using the `as` prop. This PR's primary purpose is to remedy that.
- Additional changes: migrate component to new directory structure, utilize Polymorphic (instead of Box) under the hood

🎟 _Jira ticket:_ [LG-2669](https://jira.mongodb.org/browse/LG-2669)
🎟 _Jira ticket:_ [LG-2643](https://jira.mongodb.org/browse/LG-2643)
🎟 _Jira ticket:_ [LG-2724](https://jira.mongodb.org/browse/LG-2724)

## 🛠 Types of changes

- [ ] Tooling (updates to workspace config or internal tooling)
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Style change (purely visual updates; if this is the case, attach a screenshot below!)

## ✅ Checklist

- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [x] I have run `yarn changeset` and documented my changes

### CODE REVIEW

File: packages/menu/src/Menu/Menu.types.ts [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+   * The max height of the menu (in px). Defaults to 256
+   */
+  maxHeight?: number;
+
+  /**
+   * Determines whether or not the component will be rendered in dark mode.
+   *
+   * default: `false`
+   */
+  darkMode?: boolean;
```
Comment (TheSonOfThomp) [inline packages/menu/src/Menu/Menu.types.ts:47]: Can extend `DarkModeProps`

File: packages/menu/src/MenuGroup/MenuGroup.types.ts [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+  /**
+   * Content that will appear inside of MenuGroup component.
+   * @type `<MenuItem />` | `<SubMenu />` | `<MenuGroup />` | `<MenuSeparator />`
+   */
+  children: React.ReactNode;
+
+  /**
+   * className that will be applied to root MenuGroup element.
+   */
+  className?: string;
```
Comment (TheSonOfThomp) [inline packages/menu/src/MenuGroup/MenuGroup.types.ts:11]: Shouldn't need `className`, and should extend an HTML element instead

File: packages/menu/src/MenuItem/MenuItem.types.ts [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+
+  /**
+   * Size of the MenuItem component, can be `default` or `large`
+   */
+  size?: Size;
+
+  /**
+   * className applied to  `li` element
+   */
+  className?: string;
```
Comment (TheSonOfThomp) [inline packages/menu/src/MenuItem/MenuItem.types.ts:35]: Shouldn't need `className` here either, unless we really want this TSDoc

File: packages/menu/src/SubMenu/SubMenu.tsx [hunk 4]
```diff
@@ -41,315 +39,34 @@ import {
...
-   * ```
-   * @type HTMLElement | React.Component
-   */
-  as?: React.ElementType<any>;
-}
-
-export type SubMenuProps = BoxProps<'button', BaseSubMenuProps>;
-
-const SubMenu = React.forwardRef(
+export const SubMenu = InferredPolymorphic<SubMenuProps>(
```
Comment (TheSonOfThomp) [inline packages/menu/src/SubMenu/SubMenu.tsx:69]: No default?

**Review by TheSonOfThomp — APPROVED** [review]

File: packages/menu/src/MenuGroup/MenuGroup.types.ts (hunk 2 above)
Comment (brookescarlett) [reply packages/menu/src/MenuGroup/MenuGroup.types.ts:11]: 👍

File: packages/menu/src/MenuItem/MenuItem.types.ts (hunk 3 above)
Comment (brookescarlett) [reply packages/menu/src/MenuItem/MenuItem.types.ts:35]: 👍

File: packages/menu/src/SubMenu/SubMenu.tsx (hunk 4 above)
Comment (brookescarlett) [reply packages/menu/src/SubMenu/SubMenu.tsx:69]: Good call

**Pull request merged**